python/lsst/afw/table/schema.py

```diff
@@ -27,329 +31,331 @@
 aliases = {
     str: "String",
     futurestr: "String",
-    numpy.uint16: "U",
-    numpy.int32: "I",
-    numpy.int64: "L",
-    numpy.float32: "F",
-    numpy.float64: "D",
+    np.uint16: "U",
+    np.int32: "I",
+    np.int64: "L",
+    np.float32: "F",
+    np.float64: "D",
     Angle: "Angle",
 }
 
 
-def _schemaAddField(self, field, type=None, doc="", units="",
-                    size=None, doReplace=False, parse_strict='raise'):
-    """
-    The C++ Schema class has a templated addField function, which is exposed to python via
-    Schema._addField_Suffix, where suffix is the type exposed to python, for example
-    Schema._addField_I for 32 bit integers.
-
-    This method finds the correct type for the current field and calls the appropriate C++ member.
-    """
-    # Check for astropy compatible unit string
-    astropy.units.Unit(units, parse_strict=parse_strict)
-    if type is None:
-        try:
-            prefix, suffix = __builtins__['type'](field).__name__.split("_")
-        except Exception:
-            raise TypeError("First argument to Schema.addField must be a Field if 'type' is not given.")
-        if prefix != "Field":
-            raise TypeError("First argument to Schema.addField must be a Field if 'type' is not given.")
+def addSchemaMethods(cls):
```

Comment:
Is this really needed? It is only called once. I'm not opposed to having it, but if we are changing the wrapping like this in many places (which we seem to be doing more and more) I agree with @TallJimbo that we should standardise the approach.